aws/resource_aws_elasticache_cluster.go

```diff
@@ -425,85 +427,68 @@ func resourceAwsElasticacheClusterRead(d *schema.ResourceData, meta interface{})
 	conn := meta.(*AWSClient).elasticacheconn
 	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
 
-	req := &elasticache.DescribeCacheClustersInput{
-		CacheClusterId:    aws.String(d.Id()),
-		ShowCacheNodeInfo: aws.Bool(true),
+	c, err := finder.CacheClusterWithNodeInfoByID(conn, d.Id())
+	if tfresource.NotFound(err) {
```

Comment:
Reference: https://github.com/hashicorp/terraform-provider-aws/issues/16796

```suggestion
	if !d.IsNewResource() && tfresource.NotFound(err) {
```